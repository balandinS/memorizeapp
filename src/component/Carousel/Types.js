//@flow
export type PropsItem = {
  key: string,
  title: string,
  text: string,
};

export type PropsBullet = {
  interval: number,
  items: Array<Object>,
};
