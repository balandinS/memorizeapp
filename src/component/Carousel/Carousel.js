import React, {useState} from 'react';
import {StyleSheet, Text, View, ScrollView} from 'react-native';
import {COLORS} from '../../utilities/constans';
import {widthScreen} from '../../utilities/utilities';
const ItemCarousel = (props) => {
  return (
    <View style={styles.carouselItem}>
      <View style={styles.textContainer}>
        <Text style={styles.title}>{props.title}</Text>
        <Text style={styles.text}>{props.text}</Text>
      </View>
    </View>
  );
};

// 


const Carousel = (props) => {
  const itemsPerInterval =
    props.itemsPerInterval === undefined ? 1 : props.itemsPerInterval;
  const [intervals, setIntervals] = useState(1);
  const [interval, setInterval] = useState(1);
  const [width, setWidth] = useState(0);
  const [items, setItems] = useState([
    {
      title: 'Women Collection',
      text: 'Rich selection for choosing clothes',
    },
    {
      title: 'Men Collection',
      text: 'Rich selection for choosing clothes',
    },
    {
      title: 'Child Collection',
      text: 'Rich selection for choosing clothes',
    },
  ]);

  const init = (w) => {
    setWidth(w);
    const totalItems = items.length;
    setIntervals(Math.ceil(totalItems / itemsPerInterval));
  };

  const getInterval = (offset) => {
    for (let i = 1; i <= intervals; i++) {
      if (offset < (width / intervals) * i) {
        console.log('offset < (width / intervals) --> ', intervals);
        return i;
      }
      if (i == intervals) {
        console.log('offset == (width / intervals) --> ', intervals);
        return i;
      }
    }
  };
  return (
    <View>
      <ScrollView
        horizontal={true}
        contentContainerStyle={{width: `${100 * intervals}%`}}
        showsHorizontalScrollIndicator={false}
       
        decelerationRate="fast"
        onContentSizeChange={(w, h) => {
          init(w);
        }}
        onScroll={data => {
          console.log('interval --> ', interval);
          setInterval(getInterval(data.nativeEvent.contentOffset.x));
        }}
        scrollEventThrottle={200}
        pagingEnabled>
        {items.map((item, index) => (
          <ItemCarousel
            key={index}
            title={item.title}
            text={item.text}
            isSelected={item.isSelected}
          />
        ))}
      </ScrollView>
      <View style={styles.dotContainer}>
      {items.map((item, index) => {
        return <View key={index} style={ index === interval - 1 ? selectedDot :  styles.dot} />
         
      })}
      </View>
      
    </View>
  );
};

export default Carousel;

const styles = StyleSheet.create({
  carouselItem: {
    height: 200,
    width: widthScreen,
  },
  textContainer: {
    height: 200,
    justifyContent: 'center',
  },
  title: {
    textAlign: 'center',
    fontSize: 36,
    fontWeight: 'bold',
    color: COLORS.white,
    padding: 5,
  },
  text: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '500',
    color: COLORS.white,
  },
  dotContainer: {
    height: 50,
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  dot: {
    height: 7,
    width: 7,
    borderRadius: 7 / 2,
    backgroundColor: COLORS.gray,
    margin: 5,
    opacity: .4
  },
  selectedDot: {
    backgroundColor: COLORS.white,
    transform: [{scale: 1.5}],
    opacity: .7
  },
});

const selectedDot = StyleSheet.flatten([styles.dot, styles.selectedDot]);
