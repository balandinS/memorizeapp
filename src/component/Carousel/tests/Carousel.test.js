// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from '../Carousel';

test('renders correctly', () => {
  const tree = renderer.create(<Carousel itemsPerInterval={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
