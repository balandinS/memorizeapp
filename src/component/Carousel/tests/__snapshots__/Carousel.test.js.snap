// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "width": "100%",
      }
    }
    decelerationRate="fast"
    horizontal={true}
    onContentSizeChange={[Function]}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={200}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "height": 416.6666666666667,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 416.6666666666667,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 36,
                "fontWeight": "bold",
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            Women Collection
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Rich selection for choosing clothes
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 416.6666666666667,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 416.6666666666667,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 36,
                "fontWeight": "bold",
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            Men Collection
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Rich selection for choosing clothes
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 416.6666666666667,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 416.6666666666667,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 36,
                "fontWeight": "bold",
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            Child Collection
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            Rich selection for choosing clothes
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 104.16666666666667,
        "justifyContent": "flex-end",
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "borderRadius": 3.5,
          "height": 14.583333333333334,
          "margin": 5,
          "opacity": 0.7,
          "transform": Array [
            Object {
              "scale": 1.5,
            },
          ],
          "width": 14.583333333333334,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgb(247,247,247)",
          "borderRadius": 3.5,
          "height": 14.583333333333334,
          "margin": 5,
          "opacity": 0.4,
          "width": 14.583333333333334,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgb(247,247,247)",
          "borderRadius": 3.5,
          "height": 14.583333333333334,
          "margin": 5,
          "opacity": 0.4,
          "width": 14.583333333333334,
        }
      }
    />
  </View>
</View>
`;
